scripts/generate_account.py

```diff
@@ -13,15 +13,16 @@
     metavar="KEY",
     help="Specify the private key to use instead of generating one.",
 )
+@click.option("--password", type=str, help="Choose a password to encrypt the keystore file")
 @click.argument(
     "output",
     type=click.Path(file_okay=True, dir_okay=False, path_type=Path),
 )
-def main(key: str, output: Path) -> None:
+def main(key: str, password: str, output: Path) -> None:
     if key is None:
         key = random.randbytes(32)
     account = eth_account.Account.from_key(key)
-    obj = eth_account.account.create_keyfile_json(account.key, b"")
+    obj = eth_account.account.create_keyfile_json(account.key, password)
```

Comment:
Are you sure this is going to work?
`password` is a `str`, whereas previously we gave `b""`, which is a different type. Unfortunately, there is no type checking here, but `create_keyfile_json` ultimately ends up giving the password to `hashlib.pbkdf2_hmac`, which requires that the password is a `bytes`-like object, not `str`.

Additionally, what happens if the user does not give a password? Will `click` provide `None` or and empty string?

Reply:
You are perfectly right. I just realized that the code used a method which is one abstraction layer deeper than I expected. 
`eth_account.Account.encrypt()` will do all the conversion we need and is much more convenient to use.

Regarding the password, I chose to have the password required and not giving an implicit password (i. e. `""`), to enforce the user actively typing a password.